### PR TITLE
ci: add bazel roachtest gce teamcity job

### PR DESCRIPTION
**Pull request by rickystewart**
Release note: None

### CODE REVIEW

**Comment by rickystewart** [comment]
[Sample run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/3717294).

**Review by rail — CHANGES_REQUESTED** [review]
> Reviewed 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72574)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rickystewart)

___
*[build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh, line 101 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72574#-MoJi2CqDzClPyHEPqeD:-MoJi2CqDzClPyHEPqeE:bsw7jwl) ([raw file](https://github.com/cockroachdb/cockroach/blob/adb1e029598bd72a48f19a57c17ebc56fda51021/build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh#L101)):*
> ```
> Teamcity has a 1300 minute timeout
> ```
I think you can remove this comment and `timeout`, see https://github.com/cockroachdb/cockroach/pull/72601

File: build/teamcity-nightly-roachtest.sh [hunk 1]
```diff
@@ -78,15 +78,9 @@ trap upload_stats EXIT
 # Set up the parameters for the roachtest invocation.
 PARALLELISM=16
 CPUQUOTA=1024
-ZONES=""
 TESTS=""
 case "${CLOUD}" in
   gce)
-    # We specify --zones below so that nodes are created in us-central1-b by
```
Comment (tbg) [inline build/teamcity-nightly-roachtest.sh:85]: why this change?

File: build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
```
Comment (tbg) [inline build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh:1]: This is making a copy of `teamcity-nightly-roachtest.sh` and editing it, right? I find that sort of thing fairly sad in general. I assume there are reasons, but then please outline them and add comments here and in the other copy that make it very clear that any change there also needs to be made here and vice versa.

**Review by tbg — COMMENTED** [review]
LGTM on the roachtest portion. Left some comments on the rest too but will defer to @rail for that part of the PR.

File: build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh (hunk 2 above)
Comment (rickystewart) [reply build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh:1]: I refactored out a big chunk of the script to avoid the duplication.

File: build/teamcity-nightly-roachtest.sh (hunk 1 above)
Comment (rickystewart) [reply build/teamcity-nightly-roachtest.sh:85]: This doesn't do anything. The logic to export `ZONES` was removed in https://github.com/cockroachdb/cockroach/commit/d60cf646978ceb1d4757322cdb43157681601636.

**Comment by rickystewart** [comment]
> > ```
> > Teamcity has a 1300 minute timeout
> > ```
> 
> 
> I think you can remove this comment and `timeout`, see #72601

Done.

**Review by rail — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72574#-:-Mp9_G1o2lAEuHcAtC95:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart and @tbg)

**Comment by rickystewart** [comment]
TFTR!

bors r=rail
